packages/next/client/index.tsx

```diff
@@ -629,6 +629,15 @@ function AppContainer({
   )
 }
 
+function renderApp(App: AppComponent, appProps: AppProps) {
+  if (process.env.__NEXT_RSC) {
+    const { Component, err: _, router: __, ...props } = appProps
+    return <Component {...props} />
+  } else {
+    return <App {...appProps} />
+  }
+}
+
```

Comment:
I think this breaks the incremental upgrade path: you can't create any Server Component pages without first making any existing `pages/_app` a Server Component.

That might be fine (since it's pretty trivial to do) but I'd need to spend some time thinking about any additional implications. We would probably also want a warning if you have a Client Component `pages/_app` since it'll just be ignored.